lxd/instances_post.go

```diff
@@ -957,6 +853,68 @@ func createFromBackup(s *state.State, r *http.Request, projectName string, data
 	return operations.OperationResponse(op)
 }
 
+// setupInstanceArgs sets the database instance arguments and determines the storage pool to use.
+func setupInstanceArgs(s *state.State, instType instancetype.Type, projectName string, profiles []api.Profile, req *api.InstancesPost) (storagePool string, instArgs *db.InstanceArgs, resp response.Response) {
```

Comment:
Ah this comment should become before the previous one and then the new conversion function should use it from the outset rather than duplicating it initially and then de-duping it later. Remember commits should represent final version of the change within a PR - so if you need to land some refactoring do that first before adding the feature changes.